pkg/provisioner/provisioner.go

```diff
@@ -108,10 +108,10 @@ type PrepareData struct {
 }
 
 type ServicingData struct {
-	FirmwareConfig         *metal3api.FirmwareConfig
-	TargetFirmwareSettings metal3api.DesiredSettingsMap
-	ActualFirmwareSettings metal3api.SettingsMap
-	// TargetFirmwareComponents []metal3api.FirmwareUpdate
+	FirmwareConfig           *metal3api.FirmwareConfig
+	TargetFirmwareSettings   metal3api.DesiredSettingsMap
+	ActualFirmwareSettings   metal3api.SettingsMap
+	TargetFirmwareComponents []metal3api.FirmwareUpdate
```

Comment:
Where is this field set? I think you need to update the controller to populate it somewhere.

Reply:
Now I'm puzzled, I was sure I added 
```
	hfcDirty, hfc, err := r.getHostFirmwareComponents(info)

	if err != nil {
		// wait until hostFirmwareComponents are ready
		return actionContinue{subResourceNotReadyRetryDelay}
	}
	if hfcDirty {
		prepareData.TargetFirmwareComponents = hfc.Spec.Updates
	}
```

Let me see why is not showing here 😅

Reply:
Done, and tested  with virtual BMH  =)